sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Core/BasicRetryPolicy.cs

```diff
@@ -40,6 +40,13 @@ internal class BasicRetryPolicy : EventHubsRetryPolicy
         ///
         public double JitterFactor { get; } = 0.08;
 
+        /// <summary>
+        ///   The maximum number of retry attempts before considering the associated operation
+        ///   to have failed.
+        /// </summary>
+        ///
+        public override int MaximumRetries => Options.MaximumRetries;
```

Comment:
I don't believe this is needed.    The retry policy applied to service operations should be separate from our internal "guard against the race condition" logic.

Reply:
This is on me for the bad earlier suggestion regarding retry policy use.   Sorry about that.